src/Plugins/RpcServer/RpcServer.Blockchain.cs

```diff
@@ -116,6 +118,8 @@ protected internal virtual JToken GetBlockHash(uint height)
         [RpcMethodWithParams]
         protected internal virtual JToken GetBlockHeader(BlockHashOrIndex blockHashOrIndex, bool verbose = false)
         {
+            RpcException.ThrowIfNull(blockHashOrIndex, nameof(blockHashOrIndex), RpcError.InvalidParams);
```

Comment:
maybe you can upodate the logfic of parsing the params, instead of checking in every method.